Sorbet/README.md

```diff
@@ -0,0 +1,113 @@
+Sorbet (Ruby)
+===
+
+Sorbet adds static types to Ruby.
+
+* Language resources:
+  - <https://sorbet.org/docs/overview>
+  - <https://github.com/sorbet/sorbet>
+  - <https://sorbet.org/docs/gradual>
+  - <https://sorbet.org/docs/from-typescript>
+* If-T version: **1.0**
+* Implementation: [./main.rb](./main.rb)
+* Raw command to run the benchmark: `srb tc main.rb`
+
+## Type System Basics
+
+> Q. What is the top type in this language? What is the bottom type? What is the dynamic type? If these types do not exist, explain the alternatives.
+
+* Top = `T.anything`
+* Bottom = `T.noreturn`
+* Dynamic = `T.untyped`
+
+`T.untyped` is Sorbet’s dynamic type, used for values with unknown types, such as those from untyped Ruby code. It allows any operation but provides no type safety. Sorbet supports gradual typing, so `T.untyped` is common in mixed typed/untyped codebases.
+
+<https://sorbet.org/docs/static>
+
+> Q. What base types does this implementation use? Why?
+
+`String`, `Integer`, `TrueClass`, and `FalseClass`
+
+These are standard Ruby classes, chosen for their simplicity and immutability, making them suitable for testing type narrowing in Sorbet.
```

Comment:
Can someone make a subclass of `String`? That would be a problem for type _equality_ tests.

Reply:
We ran into this issue with `type ... is` in Python and subclasses of `str`:
https://github.com/microsoft/pyright/issues/9395

One fix is to use `final` classes.

Another fix is to use `instanceof` instead of equality checks (instead of type is).

Reply:
You're probably fine with `is_a?`, and it probably is true that anyone can make a subclass of String.